feat: support hex colors with alpha

KaranUnique opened the pull request:
<!-- PR title should follow Angular Commit Message Conventions (https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines) -->

**What is the previous behavior before this PR?**

KaTeX only supported 3-digit (#RGB), 6-digit (#RRGGBB), and named colors (red, blue, etc.). 8-digit hex colors with alpha transparency (#RRGGBBAA) were rejected by the parser with "Invalid color" errors.

Example that failed before:
```latex
\textcolor{#228B2280}{F=ma}  % Error: Invalid color '#228B2280'
```

**What is the new behavior after this PR?**

KaTeX now accepts and processes 8-digit hex colors with alpha transparency in both #RRGGBBAA and RRGGBBAA formats. The alpha channel is preserved and passed through to CSS for browser rendering.

Examples that work after:
\textcolor{#FF000080}{F=ma}  % 50% transparent red
\textcolor{#228B2280}{F=ma}  % 50% transparent green  
\textcolor{00FF0040}{F=ma}   % 25% opaque green (no # prefix)

The parser now:

Validates 8-digit hex color format
Automatically adds # prefix when missing for 8-digit colors
Maintains full backward compatibility with existing color formats
Outputs standard CSS #RRGGBBAA format for modern browser alpha support


<!-- If this PR contains a breaking change, please uncomment following line -->
<!-- BREAKING CHANGE: describe its impact and migration methods -->

<!-- If this PR fixes or closes issues, please uncomment following line and change the issue number -->
Fixes #4067

edemaine commented on src/Parser.js:
```diff
@@ -680,7 +680,7 @@ export default class Parser {
             return null;
         }
-        const match = (/^(#[a-f0-9]{3}|#?[a-f0-9]{6}|[a-z]+)$/i).exec(res.text);
+        const match = (/^(#[a-f0-9]{3}|#?[a-f0-9]{6}|#?[a-f0-9]{8}|[a-z]+)$/i).exec(res.text);
```
There's also 4-digit hex codes. See https://developer.mozilla.org/en-US/docs/Web/CSS/hex-color

Is there some way to avoid the repetition here? I'm not sure.

edemaine commented on test/katex-spec.js:
```diff
@@ -912,6 +912,43 @@ describe("A color parser", function() {
+        expect(alphaColorExpression1).toParse();
+        expect(alphaColorExpression2).toParse();
+        expect(alphaColorExpression3).toParse();
+    });
```
These tests can be omitted. You're already testing parsing below, and building (which includes parsing) later.

edemaine commented on test/katex-spec.js:
```diff
@@ -912,6 +912,43 @@ describe("A color parser", function() {
+        expect(badAlphaColorExpression1).not.toParse();
+        expect(badAlphaColorExpression2).not.toParse();
+        expect(badAlphaColorExpression3).not.toParse();
+    });
```
I would inline the bad colors here, as you only use them once.

KaranUnique commented:
Hi @edemaine,

Thanks for the feedback! I’ve made the requested changes:

1.Added support for 4-digit hex colors.
2.Removed support for 8-digit hex values without a leading #.
3.Optimized the regex to avoid repetition.
4.Updated tests: removed redundant parsing tests, inlined invalid color cases, and added a 4-digit alpha color test.

I’ve pushed the updates and opened a new pull request with these fixes. Please take a look when you get a chance.
